evalml/pipelines/classification_pipeline.py

```diff
@@ -50,7 +50,7 @@ def fit(self, X, y):
     def _encode_targets(self, y):
         """Converts target values from their original values to integer values that can be processed."""
         try:
-            return pd.Series(self._encoder.transform(y))
+            return pd.Series(self._encoder.transform(y), index=y.index)
```

Comment:
Curious, did this break / how did you catch this? If so, could be good to add a test for it if there isn't one already 😬

Reply:
The time series tests caught this! We pass a `y` with a custom objective there.